distributed/tests/test_nanny.py

```diff
@@ -543,8 +543,6 @@ async def test_worker_start_exception(s):
     # ^ NOTE: `Nanny.close` sets it to `closed`, then `Server.start._close_on_failure` sets it to `failed`
     assert nanny.process is None
     assert "Restarting worker" not in logs.getvalue()
-    # Avoid excessive spewing. (It's also printed once extra within the subprocess, which is okay.)
-    assert logs.getvalue().count("ValueError: broken") == 1, logs.getvalue()
```

Comment:
Previously we were logging the exception twice. Once in the subprocess, once in the parent process.
Now, we're only _logging_ it once in the subprocess and are raising it to the caller as a proper exception which should make for a nicer UX

logs before

```
2022-11-25 10:51:13,721 - distributed.scheduler - INFO - State start
2022-11-25 10:51:13,723 - distributed.scheduler - INFO -   Scheduler at:     tcp://127.0.0.1:64939
2022-11-25 10:51:13,724 - distributed.scheduler - INFO -   dashboard at:           127.0.0.1:64938
2022-11-25 10:51:14,277 - distributed.worker - INFO - Stopping worker. Reason: worker-close
2022-11-25 10:51:14,277 - distributed.worker - INFO - Closed worker has not yet started: Status.init
2022-11-25 10:51:14,278 - distributed.nanny - ERROR - Failed to start worker
Traceback (most recent call last):
  File "/Users/fjetter/workspace/distributed/distributed/core.py", line 482, in start
    await asyncio.wait_for(self.start_unsafe(), timeout=timeout)
  File "/Users/fjetter/mambaforge/envs/dask-distributed-310/lib/python3.10/asyncio/tasks.py", line 408, in wait_for
    return await fut
  File "/Users/fjetter/workspace/distributed/distributed/tests/test_nanny.py", line 527, in start_unsafe
    raise ValueError("broken")
ValueError: broken

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/fjetter/workspace/distributed/distributed/nanny.py", line 921, in run
    async with worker:
  File "/Users/fjetter/workspace/distributed/distributed/core.py", line 495, in __aenter__
    await self
  File "/Users/fjetter/workspace/distributed/distributed/core.py", line 490, in start
    raise RuntimeError(f"{type(self).__name__} failed to start.") from exc
RuntimeError: BrokenWorker failed to start.
2022-11-25 10:51:14,314 - distributed.nanny - ERROR - Failed to start process
Traceback (most recent call last):
  File "/Users/fjetter/workspace/distributed/distributed/core.py", line 482, in start
    await asyncio.wait_for(self.start_unsafe(), timeout=timeout)
  File "/Users/fjetter/mambaforge/envs/dask-distributed-310/lib/python3.10/asyncio/tasks.py", line 408, in wait_for
    return await fut
  File "/Users/fjetter/workspace/distributed/distributed/tests/test_nanny.py", line 527, in start_unsafe
    raise ValueError("broken")
ValueError: broken

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/fjetter/workspace/distributed/distributed/nanny.py", line 442, in instantiate
    result = await self.process.start()
  File "/Users/fjetter/workspace/distributed/distributed/nanny.py", line 714, in start
    msg = await self._wait_until_connected(uid)
  File "/Users/fjetter/workspace/distributed/distributed/nanny.py", line 857, in _wait_until_connected
    raise msg["exception"]
  File "/Users/fjetter/workspace/distributed/distributed/nanny.py", line 921, in run
    async with worker:
  File "/Users/fjetter/workspace/distributed/distributed/core.py", line 495, in __aenter__
    await self
  File "/Users/fjetter/workspace/distributed/distributed/core.py", line 490, in start
    raise RuntimeError(f"{type(self).__name__} failed to start.") from exc
RuntimeError: BrokenWorker failed to start.
2022-11-25 10:51:14,318 - distributed.scheduler - INFO - Scheduler closing...
2022-11-25 10:51:14,319 - distributed.scheduler - INFO - Scheduler closing all comms
```


logs after

```
2022-11-25 10:50:08,962 - distributed.scheduler - INFO - State start
2022-11-25 10:50:08,963 - distributed.scheduler - INFO -   Scheduler at:     tcp://127.0.0.1:64916
2022-11-25 10:50:08,963 - distributed.scheduler - INFO -   dashboard at:           127.0.0.1:64915
2022-11-25 10:50:09,388 - distributed.worker - INFO - Stopping worker. Reason: worker-close
2022-11-25 10:50:09,388 - distributed.worker - INFO - Closed worker has not yet started: Status.init
2022-11-25 10:50:09,390 - distributed.nanny - ERROR - Failed to start worker
Traceback (most recent call last):
  File "/Users/fjetter/workspace/distributed/distributed/core.py", line 487, in start
    await asyncio.wait_for(self.start_unsafe(), timeout=timeout)
  File "/Users/fjetter/mambaforge/envs/dask-distributed-310/lib/python3.10/asyncio/tasks.py", line 408, in wait_for
    return await fut
  File "/Users/fjetter/workspace/distributed/distributed/tests/test_nanny.py", line 527, in start_unsafe
    raise ValueError("broken")
ValueError: broken

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/fjetter/workspace/distributed/distributed/nanny.py", line 899, in run
    async with worker:
  File "/Users/fjetter/workspace/distributed/distributed/core.py", line 501, in __aenter__
    await self
  File "/Users/fjetter/workspace/distributed/distributed/core.py", line 497, in start
    raise RuntimeError(f"{type(self).__name__} failed to start.") from exc
RuntimeError: BrokenWorker failed to start.
2022-11-25 10:50:09,439 - distributed.scheduler - INFO - Scheduler closing...
2022-11-25 10:50:09,440 - distributed.scheduler - INFO - Scheduler closing all comms
```

As you can see, only logging in the subprocess gives also a nicer traceback than both